base/server/src/main/java/com/netscape/cmscore/dbs/DBSearchResults.java

```diff
@@ -66,14 +66,13 @@ public Object nextElement() {
             if (o instanceof LDAPEntry) {
                 entry = (LDAPEntry) o;
                 return mRegistry.createObject(entry.getAttributeSet());
-            } else {
-                if (o instanceof LDAPException)
-                    ;
-                // doing nothing because the last object in the search
-                // results is always LDAPException
-                else
-                    logger.warn("DBSearchResults: result format error class=" + o.getClass().getName());
             }
+            if (o instanceof LDAPException)
```

Comment:
Why not invert and print the warning if it is not an `LDAException`

Reply:
I will push an additional fix (there is other stuff to clean up here).